World Air Quality Index sensor

valentinalexeev opened the pull request:
**Description:**
Implementation of a sensor that queries World Air Quality Index service for data at specific locations.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1451

**Example entry for `configuration.yaml` (if applicable):**
```yaml
# Example configuration.yaml entry
sensor:
  - platform: waqi
    locations:
      - beijin
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - New dependencies have been added to the `REQUIREMENTS` variable.
  - New dependencies are only imported inside functions that use them.
  - New dependencies have been added to `requirements_all.txt`.
  - New files were added to `.coveragerc`.

valentinalexeev committed:
Implement WAQI sensor

valentinalexeev committed:
Corrections based on CI check.

valentinalexeev committed:
Updated requirements_all.txt for pwaqi==1.2

valentinalexeev commented:
Please do not merge it as of yet. There is still one issue with the pwaqi component that prevents the sensor from getting current data.

valentinalexeev committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into feature/WAQI_Sensor

valentinalexeev committed:
Require latest version of pwaqi

valentinalexeev committed:
Fix lint: single argument for .exception and no more pass statement.

valentinalexeev committed:
Further lint fixes.

valentinalexeev commented:
With pwaqi updated to 1.2 it works fine.

Can someone explain the reason of failed builds on travis? Tox runs fine for me in a local environment.

valentinalexeev committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into feature/WAQI_Sensor

valentinalexeev committed:
Merge branch 'dev' of https://github.com/valentinalexeev/home-assistant into feature/WAQI_Sensor

valentinalexeev committed:
pydocstyle fix

balloob commented on .coveragerc:
```diff
@@ -308,6 +308,7 @@ omit =
     homeassistant/components/sensor/vasttrafik.py
     homeassistant/components/sensor/xbox_live.py
     homeassistant/components/sensor/yweather.py
+    homeassistant/components/sensor/waqi.py
```
please sort alphabetically

balloob commented on homeassistant/components/sensor/waqi.py:
```diff
@@ -0,0 +1,85 @@
+Support for the World Air Quality Index service.
+
+For more details about this platform, please refer to the documentation at
+https://github.com/home-assistant/home-assistant/
```
Wrong url.

balloob commented on homeassistant/components/sensor/waqi.py:
```diff
@@ -0,0 +1,85 @@
+    dev = []
+    try:
+        import pwaqi
+        _LOGGER.info('Known locations: %s', config[ATTR_LOCATION])
```
These should not be `info`. `debug` at best but to be honest, this is a lot of logging.

balloob commented on homeassistant/components/sensor/waqi.py:
```diff
@@ -0,0 +1,85 @@
+            _LOGGER.info('I got the following stations: %s', station_ids)
+            for station in station_ids:
+                dev.append(WaqiSensor(station))
+    except KeyError:
```
Instead of expecting a key error, include a `PLATFORM_SCHEMA` that will enforce that the key is present.

balloob commented on homeassistant/components/sensor/waqi.py:
```diff
@@ -0,0 +1,85 @@
+        """Return the unit of measurement of this entity, if any."""
+        return "AQI"
+
+    def update(self):
```
Add `@Throttle(MIN_TIME_BETWEEN_UPDATES)` or else no throttling will be applied and it will be queried every 15 seconds.

valentinalexeev replied:
Thanks! That was my next fix already :)
Done in 2b29d64

valentinalexeev committed:
Implement rate throttle.
Data on WAQI is usually updated once an hour - make it refresh every thirty minutes.

valentinalexeev committed:
Implement schema validation with voluptuous.
Change exception handling scope.
Move messages to debug().

valentinalexeev committed:
Fix lint (empty indented line).

valentinalexeev committed:
Sort lines correctly.

valentinalexeev committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into feature/WAQI_Sensor

valentinalexeev committed:
Fix last lint issue.

valentinalexeev commented:
@balloob can you check it once again? It seems that I had implemented all your corrections :)

valentinalexeev committed:
Provide additional sensor data as received from WAQI.
Easier-to-read throttle timing.

valentinalexeev committed:
Additional object attributes to be unrolled later.

balloob commented:
Awesome 🐬
